service/src/client.rs

```diff
@@ -90,7 +105,15 @@ impl<S: PostService> ServiceClient<S> {
                         .identity(identity),
                 )?
             }
-            None => endpoint,
+            None => {
+                if scheme.as_str() == "https" {
```

Comment:
NIT: it's little less error-prone:
```suggestion
                if scheme == Scheme::HTTPS {
```